### PR TITLE
[TFF-189] Return item

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary

Add isCheckedOutTo attribute to items.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Review Code
2. Look at client PR

### Links

- [TFF-189](https://msoese.atlassian.net/browse/TFF-189)
- TheFourFifths/consus-client/pull/104

### CODE REVIEW

File: docs/REST.md [hunk 1]
```diff
@@ -57,7 +58,8 @@ Retrieve an item.
         "item": {
             "address": "iGwEZUvfA",
             "modelAddress": "m8y7nEtAe",
-            "status": "AVAILABLE"
+            "status": "AVAILABLE",
+            "isCheckedOutTo": "student.id"
```
Comment (thofmann) [inline docs/REST.md]: Nit: Please use an actual student ID for the examples, e.g. `123456`.

Also, this should be a number and not a string, shouldn't it?

File: docs/REST.md [hunk 2]
```diff
@@ -364,7 +366,8 @@ Submit a check-in request.
     "status": "success",
     "data": {
         "itemAddress": "iGwEZUvfA",
-        "modelName": "Resistor"
+        "modelName": "Resistor",
+        "isCheckedOutTo": "student.id"
```
Comment (thofmann) [inline docs/REST.md]: Same as above.

File: src/store/item-store.js [hunk 3]
```diff
@@ -70,7 +71,7 @@ store.registerHandler('NEW_CHECKOUT', data => {
     store.waitFor(CheckoutStore);
     data.itemAddresses.forEach(address => {
         store.getItemByAddress(address).status = 'CHECKED_OUT';
-
+        store.getItemByAddress(address).isCheckedOutTo = data.studentId;
```
Comment (thofmann) [inline src/store/item-store.js]: Nit: This won't work when we start using `clone`.

File: src/store/item-store.js [hunk 4]
```diff
@@ -91,6 +92,7 @@ store.registerHandler('CHECKIN', data => {
         return;
     }
     store.getItemByAddress(data.itemAddress).status = 'AVAILABLE';
+    store.getItemByAddress(data.itemAddress).isCheckedOutTo = undefined;
```
Comment (thofmann) [inline src/store/item-store.js]: NACK: `null` or `undefined`? I think `null` is being used in the other places.

File: docs/REST.md (hunk 1 above)
Comment (gearedmonkey) [inline docs/REST.md]: changed to a number

File: docs/REST.md (hunk 2 above)
Comment (gearedmonkey) [inline docs/REST.md]: Changed to a number

File: src/store/item-store.js (hunk 4 above)
Comment (gearedmonkey) [inline src/store/item-store.js]: Should be null. I had switched between the two and missed this one. Changed.

**Pull request merged**